src/main/java/org/opensrp/web/rest/StockResource.java

```diff
@@ -203,4 +218,111 @@ public Stock update(Stock stock) {
 		return stockService.findAllStocks();
 	}
 
+	@PostMapping(consumes = { MediaType.APPLICATION_JSON_VALUE,
+			MediaType.TEXT_PLAIN_VALUE })
+	public ResponseEntity<String> create(@RequestBody Inventory inventory) {
+		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+		String userName = authentication.getName();
+		stockService.addInventory(inventory, userName);
+		return new ResponseEntity<>(HttpStatus.CREATED);
+	}
+
+	@PutMapping(value = "/{id}", consumes = { MediaType.APPLICATION_JSON_VALUE,
+			MediaType.TEXT_PLAIN_VALUE })
+	public ResponseEntity<String> update(@PathVariable("id") String stockId, @RequestBody Inventory inventory) {
+		if (stockId == null) {
+			return new ResponseEntity<>("Stock item id is required", RestUtils.getJSONUTF8Headers(),
+					HttpStatus.BAD_REQUEST);
+		} else {
+			Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+			String userName = authentication.getName();
+			stockService.updateInventory(inventory, userName);
+			return new ResponseEntity<>(HttpStatus.CREATED);
+		}
+	}
+
+	@DeleteMapping(value = "/{id}", produces = { MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<String> delete(@PathVariable(Constants.RestPartVariables.ID) Long id) {
+		if (id == null) {
+			return new ResponseEntity<>("Stock item id is required", RestUtils.getJSONUTF8Headers(),
+					HttpStatus.BAD_REQUEST);
+		} else {
+			stockService.deleteStock(id);
+			return new ResponseEntity<>("Stock deleted successfully", RestUtils.getJSONUTF8Headers(),
+					HttpStatus.NO_CONTENT);
+		}
+	}
+
+	@GetMapping(value = "/servicePointId/{servicePointId}", produces = {
+			MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<List<Stock>> getStockItemsByServicePoint(@PathVariable String servicePointId) {
+		return new ResponseEntity<>(stockService.getStocksByServicePointId(servicePointId),
+				RestUtils.getJSONUTF8Headers(),
+				HttpStatus.OK);
+	}
+
+	@PostMapping(headers = { "Accept=multipart/form-data" }, produces = {
+			MediaType.APPLICATION_JSON_VALUE }, value = "/import/inventory")
+	public ResponseEntity importInventoryData(@RequestParam("file") MultipartFile file)
+			throws IOException {
+		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+		String userName = authentication.getName();
+		List<Map<String, String>> csvClients = readCSVFile(file);
+		CsvBulkImportDataSummary csvBulkImportDataSummary = stockService
+				.convertandPersistInventorydata(csvClients, userName);
+
+		generateCSV(csvBulkImportDataSummary);
+		File csvFile = new File(SAMPLE_CSV_FILE);
```

Comment:
Always generate a new CSV every time so that there is conflict of sessions when users are uploading concurrenlly